devito/ops/types.py

```diff
@@ -31,13 +31,16 @@ class OpsAccessible(basic.Symbol):
     """
     is_Scalar = True
 
-    def __new__(cls, name, dtype, read_only, *args, **kwargs):
+    def __new__(cls, name, dtype, read_only, origin, origin_time_index, * args, **kwargs):
         obj = basic.Symbol.__new__(cls, name, dtype, *args, **kwargs)
-        obj.__init__(name, dtype, read_only, *args, **kwargs)
+        obj.__init__(name, dtype, read_only, origin, origin_time_index, *args, **kwargs)
```

Comment:
Im not a big fan of adding this sort of information to a Symbol....it really feels odd. Can't the caller just retain this info, perhaps in a dict?

Reply:
That's a good idea. 
I try the dictionary approach.

Reply:
I tried creating a dictionary approach... but it didn't feel right.

I think this `OpsAccessible` is much like the `Indexed` class, as it also needs information of what it is accessing (in this case TimeFunction or Function). But, to not store all the data from `TimeFunction`, I'll try to just use the necessary information from them. 

I'll commit and then you can see if this is acceptable.

Reply:
OK, I see